lib/src/fsmonitor.rs

```diff
@@ -153,7 +169,14 @@ pub mod watchman {
         /// copy to build up its in-memory representation of the
         /// filesystem, which may take some time.
         #[instrument]
-        pub async fn init(working_copy_path: &Path) -> Result<Self, Error> {
+        pub async fn init(
+            kind: super::FsmonitorKind,
+            working_copy_path: &Path,
+        ) -> Result<Self, Error> {
+            let super::FsmonitorKind::Watchman { set_up_trigger } = kind else {
+                panic!("Unexpected FsmonitorKind: {:?}", kind);
+            };
```

Comment:
suggestion: I would rather that this not be fallible; either pass the individual arguments in (`set_up_trigger`) as parameters, or create some kind of `WatchmanOptions` or `WatchmanConfig` `struct` to aggregate the options and pass that in.

suggestion: In either case, put the "option arguments" after the "object arguments", i.e. put `working_copy_path` first.

Reply:
I like that.  Done